ui/src/lib/useChatStore.ts

```diff
@@ -88,13 +88,15 @@ const useChatStore = create<ChatState>((set, get) => ({
       return;
     }
 
-    if (
+    // Keep status as "thinking" if we receive any tool-related messages
+    const isToolMessage = 
       messageUtils.isTeamResult(messageConfig) ||
       messageUtils.isFunctionExecutionResult(messageConfig.content) ||
       messageUtils.isToolCallContent(messageConfig.content) ||
       messageUtils.isMultiModalContent(messageConfig.content) ||
-      messageUtils.isLlmCallEvent(messageConfig.content)
-    ) {
+      messageUtils.isLlmCallEvent(messageConfig.content);
+
+    if (isToolMessage) {
```

Comment:
I tried this out, but the second time I send a message (that doesn't invoke the function/tool), I still see "ready" status (and it should be "thinking").